src/lib/get-file.js

```diff
@@ -5,31 +5,65 @@ const mime = require('mime');
 
 const getPathByBasePath = require('./paths').getPathByBasePath;
 
+const readFileAsync = (filePath) => {
+    return new Promise((resolve, reject) => {
+        fs.readFile(filePath, (err, data) => {
+            if (err) {
+                reject(err);
+            }
+            resolve(data);
+        });
+    });
+};
+
+const existFileAsync = (filePath) => {
+    return new Promise((resolve, reject) =>
+        fs.access(filePath, (err) => {
+            if (err) {
+                reject();
+            }
+            resolve(filePath);
+        })
+    );
+};
+
+const oneSuccess = (promises) => {
+    return Promise.all(promises.map((p) => {
+        return p.then(
+            (val) => Promise.reject(val),
+            (err) => Promise.resolve(err)
+        );
+    })).then(
+        (errors) => Promise.reject(errors),
+        (val) => Promise.resolve(val)
+    );
+};
+
 /**
  *
  * @param {PostcssUrl~Asset} asset
  * @param {PostcssUrl~Options} options
  * @param {PostcssUrl~Dir} dir
  * @param {Function} warn
- * @returns {PostcssUrl~File}
+ * @returns {Promise<PostcssUrl~File | Undefined>}
  */
 const getFile = (asset, options, dir, warn) => {
     const paths = options.basePath
         ? getPathByBasePath(options.basePath, dir.from, asset.pathname)
         : [asset.absolutePath];
-    const filePath = paths.find(fs.existsSync);
-
-    if (!filePath) {
-        warn(`Can't read file '${paths.join()}', ignoring`);
 
-        return;
-    }
+    return oneSuccess(paths.map((path) => existFileAsync(path)))
```

Comment:
```js
const findExistsPath = (paths) => {
    let resolved = false;

    return new Promise((resolve, reject) =>
        const findPromises = paths.map(path => {
            return existFileAsync(path).then((isExists) => { 
               !resolved && isExists && resolve(path);
               resolved = true;
            })
        })

        Promise.all(findPromises).then(reject);
    );
}
```

Reply:
`Promise.all` waits until all promises are fulfilled. it is not necessary